flytekit/annotated/testing.py

```diff
@@ -35,3 +37,28 @@ def _log(*args, **kwargs):
     t.execute = _log
     yield m
     t.execute = _captured_fn
+
+
+def patch(target: Union[PythonTask, Workflow]):
+    if not isinstance(target, PythonTask) and not isinstance(target, Workflow):
```

Comment:
like this type-check. Can you add it to `task_mock` as well?

Reply:
sure.